src/lib/firestore-schema.ts

```diff
@@ -0,0 +1,152 @@
+
+/*
+ * classes
+ *
+ * Subcollection of domain with several fields copied from portal (e.g. name, teachers) with
+ * potential synchronization issues.
+ * Class resources are accessible to other teachers that share a network with one of the teachers of the class.
+ * Networked access will be mediated by a Firebase function which can check whether the requesting user and
+ * one of the teachers of the class share a network.
+ */
+interface ClassDocument {
+  id: string;                 // portal class id
+  name: string;               // portal class name
+  hash: string;               // portal class hash
+  teachers: string[];         // uids of teachers of class
+}
+// collection key is class id (or class hash?)
+type ClassesCollection = FSCollection<ClassDocument>;
+
+/*
+ * networks
+ *
+ * Subcollection of domain with very little in it. We could try to store a list of users or classes in network
+ * but this would just create synchronization headaches. It seems better to query for those things, at least
+ * initially. I considered making some collections like classes subcollections of networks, but in each case
+ * it seemed like there might be value in being able to access them without going through the network.
+ */
+interface NetworkDocument {
+  id: string;                 // admin-defined, e.g. "msu-1", because they will need to be hand-entered initially
+  name: string;               // the user-visible full name of the network
+  shortName: string;          // abbreviated name for use in compact displays
+}
+// collection key is id, e.g. "msu-1"
+type NetworksCollection = FSCollection<NetworkDocument>;
+
+/*
+ * mcsupports
+ *
+ * Preexisting subcollection of domain used to implement multi-class supports.
+ */
+interface SupportDocument {
+  classes: string[];          // array of class hashes that can access the support
+  content: string;            // JSON stringified content of support
+  context_id: string;         // class hash of class in which support was created
+  createdAt: FSDate;          // timestamp
+  originDoc: string;          // id of document from which the support was published
+  platform_id: string;        // portal, e.g. "learn.concord.org"
+  problem: string;            // e.g. "msa/2/1"
+  properties: {
+    caption: string;          // e.g. "MSA 2.1 Problem Title"
+    teacherSupport: boolean;
+  };
+  resource_link_id: string;   // offering id
+  resource_url: string;       // e.g. "https://collaborative-learning.concord.org/?unit=msa&problem=2.1"
+  type: string;               // e.g. "supportPublication"
+  uid: string;                // user id of teacher that created the support
+}
+type SupportsCollection = FSCollection<SupportDocument>;
+
+/*
+ * A domain represents the set of resources available within a specific domain.
+ * All access is restricted by the domain, i.e. users in one domain can only access resources in that domain.
+ */
+interface DomainDocument {
```

Comment:
This kind of corresponds to the platform_id that I was proposing in the other comments. But I see there is just single domain for authed. There might multiple authed domains. 

I'm not sure the right solution. Perhaps there is just a collection of domains. And each domain can have a name and platform_id.

Reply:
As mentioned above, for good or ill this is part of the existing Firestore database implementation in which `authed` is a subcollection, and `learn_concord_org` and `learn_staging_concord_org` are `DomainDocument`s within the `authed` subcollection. As indicated, all of the other subcollections are underneath a particular `DomainDocument`.

Reply:
Sounds good.